docs: Inform multiple region support not an option for iam-role-eks (#299)
terraform-aws-modules/terraform-aws-iam: modules/iam-eks-role/README.md

- Support multiple `ServiceAccount` in the same cluster, for example when a workload runs in multiple namespaces
- More suitable for non-cluster admins as implementation is simpler
- More suitable for when the IAM role and cluster resources are in separate Terraform configurations

> - Does not support multiple regions, consider using [iam-assumable-role-with-oidc](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-assumable-role-with-oidc/) if role needs to support clusters in multiple regions

Reviewer: ```suggestion
- Does not support multiple regions, consider using [iam-role-for-service-accounts-eks](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-role-for-service-accounts-eks/) if role needs to support clusters in multiple regions
```

> - Does not support multiple regions, consider using [iam-assumable-role-with-oidc](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-assumable-role-with-oidc/) if role needs to support clusters in multiple regions

Reviewer: Can we update/combine this with the 2nd bullet since I think they are related. It supports multiple clusters, but the caveat is that it does not support multiple regions due to the use of the `aws_eks_cluster` data source

```suggestion
- Does not support multiple regions, consider using [iam-role-for-service-accounts-eks](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-role-for-service-accounts-eks/) if role needs to support clusters in multiple regions
```

cc @max-rocket-internet

Reviewer 2: Hey 👋

@sepulworld 

>  inform user that this sub module will not work for multiple regions

There is a note already in the [README](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-eks-role/README.md) that says "The EKS cluster needs to exist first, in the current AWS account and region". I think that's pretty clear, no?

> consider using iam-role-for-service-accounts-eks

I don't think this is great advice since the `iam-role-for-service-accounts-eks` module is stated for "commonly used controllers/custom resources" (quite specific) whereas `iam-role-eks` has no specific use case like this, it's simply to create a general IRSA role that could be used for anything. The existing readme already details differences compared to [iam-assumable-role-with-oidc](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-assumable-role-with-oidc/), perhaps we recommend using this for multi-region?
